client/src/pages/admin/mentor-registry.tsx

```diff
@@ -46,17 +60,21 @@ function Page(props: Props) {
   const loadData = useCallback(
     withLoading(async () => {
       const [allData, courses] = await Promise.all([mentorRegistryService.getMentors(), coursesService.getCourses()]);
+      const { data: disciplines } = await disciplinesApi.getDisciplines();
       setAllData(allData);
       updateData(showAll, allData);
       setCourses(courses);
+      setDisciplines(disciplines);
     }),
     [showAll],
   );
 
   const cancelMentor = useCallback(
```

Comment:
I'd propose to add proper deps array or remove `useCallback` here

Reply:
fixed